pkg/cmd/tasks/initcmd/init.go

```diff
@@ -6,112 +6,150 @@ package initcmd
+	"github.com/airplanedev/cli/pkg/cmd/tasks/initcmd/initcompat"
+	"github.com/airplanedev/cli/pkg/fs"
 	"github.com/airplanedev/cli/pkg/logger"
+	"github.com/airplanedev/cli/pkg/runtime"
+	_ "github.com/airplanedev/cli/pkg/runtime/javascript"
+	_ "github.com/airplanedev/cli/pkg/runtime/typescript"
 	"github.com/airplanedev/cli/pkg/utils"
-	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 type config struct {
-	root *cli.Config
-	file string
-	from string
+	client *api.Client
+	file   string
+	slug   string
+	from   string
 }
 
 func New(c *cli.Config) *cobra.Command {
-	var cfg = config{root: c}
+	var cfg = config{client: c.Client}
 
 	cmd := &cobra.Command{
 		Use:   "init",
 		Short: "Initialize a task definition",
 		Example: heredoc.Doc(`
 			$ airplane tasks init
+			$ airplane tasks init --slug task-slug ./my/task.js
+			$ airplane tasks init --slug task-slug ./my/task.ts
 			$ airplane tasks init -f ./airplane.yml
 			$ airplane tasks init --from hello_world
 		`),
+		Args: cobra.RangeArgs(0, 1),
 		PersistentPreRunE: utils.WithParentPersistentPreRunE(func(cmd *cobra.Command, args []string) error {
 			return login.EnsureLoggedIn(cmd.Root().Context(), c)
 		}),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return run(cmd.Root().Context(), cfg)
+			var ctx = cmd.Root().Context()
+
+			if cfg.slug != "" && len(args) == 0 {
+				return errors.New("<path> is required when providing a --slug")
+			}
+
+			if len(args) == 1 {
+				if cfg.slug == "" {
+					return errors.New("--slug is required when providing a <path>")
+				}
+				cfg.file = args[0]
+				return run(ctx, cfg)
+			}
+
+			// compat
+			return initcompat.Run(ctx, initcompat.Config{
+				Client: cfg.client,
+				From:   cfg.from,
+				File:   cfg.file,
+			})
 		},
 	}
 
-	cmd.Flags().StringVarP(&cfg.file, "file", "f", "", "Path to a file to store task definition")
+	cmd.Flags().StringVar(&cfg.slug, "slug", "", "Slug of an existing task to generate from.")
 	cmd.Flags().StringVar(&cfg.from, "from", "", "Slug of an existing task to generate from")
+	cmd.Flags().StringVarP(&cfg.file, "file", "f", "", "Path to a file to store task definition")
 
 	return cmd
 }
 
 func run(ctx context.Context, cfg config) error {
-	var kind initKind
-	var err error
-	// If --from is provided, we already know the user wants to create
-	// from an existing task, so we don't need to prompt the user here.
-	if cfg.from == "" {
-		logger.Log("Airplane is a development platform for engineers building internal tools.\n")
-		logger.Log("This command will configure a task definition which Airplane uses to deploy your task.\n")
-
-		if kind, err = pickInitKind(); err != nil {
-			return err
-		}
-	} else {
-		kind = initKindTask
+	var client = cfg.client
+
+	task, err := client.GetTask(ctx, cfg.slug)
+	if err != nil {
+		return err
+	}
+
+	r, ok := runtime.Lookup(cfg.file)
+	if !ok {
+		return fmt.Errorf("cannot find a runtime that matches %s", cfg.file)
```

Comment:
Maybe something like this?

```suggestion
		return fmt.Errorf("Unable to deploy tasks with a %s file ending.", fileEnding)
```

Reply:
What if you do `init --slug /tmp/foo` then there's no file ext 😅

Reply:
That could be a separate error! `Expected a file ending` etc.